### PR TITLE
IDEMPIERE-3981: FindWindow - Overwrite isSelection at field level, Va…

### PR DESCRIPTION
**Pull request by dpansheriya**
…lidation Rule overwrite for Find window, Extended search and advanced hiding options.

@globalqss 
The Patch was originally submitted by Murilo. We have reviewed and cleaned code and tested.
from AD_Field, AD_Field_VT, AD_Tab_V, AD_Tab_Vt and AD_Window_VT dropped entityType as not related to this enhancement.

### CODE REVIEW

**Comment by hengsin** [comment]
will be great if we can take this opportunitiy to add unit test cases to validate the various view (ad_field_v, ad_field_vt, etc).

**Comment by hengsin** [comment]
a note for concern of performance - rather than making advance search and fields not visible, a more practical approach is to have query timeout for both simple and advance find.

**Comment by dpansheriya** [comment]
> a note for concern of performance - rather than making advance search and fields not visible, a more practical approach is to have query timeout for both simple and advance find.

Thanks @hengsin for your feedback. Just want to clarify the meaning of time out.
Do you mean here that field should not be loaded, correct?

**Comment by hengsin** [comment]
Hi @dpansheriya , https://docs.oracle.com/en/java/javase/11/docs/api/java.sql/java/sql/Statement.html#setQueryTimeout(int)

**Comment by dpansheriya** [comment]
> Hi @dpansheriya , https://docs.oracle.com/en/java/javase/11/docs/api/java.sql/java/sql/Statement.html#setQueryTimeout(int)

@hengsin Thanks and now I understood your comment. This enhancement is not only targeting performance but also data security issue. Current advanced filtering let you build complex  filter which may not advisable in some implementation as per the discussion on ticket.

File: migration/i7.1z/oracle/202009111500_IDEMPIERE-3981.sql [hunk 1]
```diff
@@ -0,0 +1,1036 @@
...
+-- Jun 22, 2019, 3:57:01 PM BRT
+UPDATE AD_Field SET EntityType='D', AD_Reference_Value_ID=NULL, AD_Val_Rule_ID=NULL, IsToolbarButton=NULL,Updated=TO_DATE('2019-06-22 15:57:01','YYYY-MM-DD HH24:MI:SS'),UpdatedBy=100 WHERE AD_Field_ID=206124
+;
+
+-- Jun 22, 2019, 3:57:10 PM BRT
+UPDATE AD_Field SET AD_Reference_Value_ID=NULL, AD_Val_Rule_ID=NULL, IsToolbarButton=NULL,Updated=TO_DATE('2019-06-22 15:57:10','YYYY-MM-DD HH24:MI:SS'),UpdatedBy=100 WHERE AD_Field_ID=206126
+;
+
+-- Jun 22, 2019, 3:57:30 PM BRT
+UPDATE AD_Field SET SeqNo=360,IsDisplayed='Y', Updated=statement_timestamp(), UpdatedBy=100 WHERE AD_Field_ID=206126
```
Comment (globalqss) [inline migration/i7.1z/oracle/202009111500_IDEMPIERE-3981.sql:541]: statement_timestamp() is not valid in oracle

File: migration/i7.1z/postgresql/202009111500_IDEMPIERE-3981.sql [hunk 2]
```diff
@@ -0,0 +1,1046 @@
...
+    t.AD_ColumnSortOrder_ID, t.AD_ColumnSortYesNo_ID, t.Included_Tab_ID, t.Parent_Column_ID,
+    t.AD_Tab_UU, tbl.AD_Table_UU, t.TreeDisplayedOn, t.MaxQueryRecords,
+    t.IsAllowAdvancedLookup, t.IsLookupOnlySelection
+FROM AD_Tab t 
+	INNER JOIN AD_Table tbl ON (t.AD_Table_ID = tbl.AD_Table_ID)
+	INNER JOIN AD_Tab_Trl trl ON (t.AD_Tab_ID = trl.AD_Tab_ID)
+WHERE t.IsActive='Y'
+  AND tbl.IsActive='Y'
+;
+
```
Comment (globalqss) [inline migration/i7.1z/postgresql/202009111500_IDEMPIERE-3981.sql:1046]: postgres script requires:
 SELECT register_migration_script('202009111500_IDEMPIERE-3981.sql') FROM dual 
same as the oracle script

Comment (dpansheriya) [reply migration/i7.1z/postgresql/202009111500_IDEMPIERE-3981.sql:1046]: This is chilly mistake, Let me fix.

File: migration/i7.1z/oracle/202009111500_IDEMPIERE-3981.sql (hunk 1 above)
Comment (dpansheriya) [reply migration/i7.1z/oracle/202009111500_IDEMPIERE-3981.sql]: Fixed by using updated time of migration script generation

**Comment by dpansheriya** [comment]
Both review comments are fixed.

File: migration/i7.1z/oracle/202009111500_IDEMPIERE-3981.sql [hunk 3]
```diff
@@ -0,0 +1,1036 @@
...
+   FROM ad_field f
+   JOIN ad_field_trl trl ON f.ad_field_id = trl.ad_field_id
+   JOIN ad_tab t ON f.ad_tab_id = t.ad_tab_id
+   LEFT JOIN ad_fieldgroup fg ON f.ad_fieldgroup_id = fg.ad_fieldgroup_id
+   LEFT JOIN ad_fieldgroup_trl fgt ON f.ad_fieldgroup_id = fgt.ad_fieldgroup_id AND trl.ad_language = fgt.ad_language
+   LEFT JOIN ad_column c ON f.ad_column_id = c.ad_column_id
+   LEFT JOIN ad_column_trl ct ON f.ad_column_id = ct.ad_column_id AND trl.ad_language = ct.ad_language
+   JOIN ad_table tbl ON c.ad_table_id = tbl.ad_table_id
+   LEFT JOIN ad_val_rule vr ON vr.ad_val_rule_id = COALESCE(f.ad_val_rule_id, c.ad_val_rule_id)
+   LEFT JOIN ad_val_rule vrf ON vrf.ad_val_rule_id = COALESCE(f.ad_val_rule_Lookup_id, c.ad_val_rule_Lookup_id
```
Comment (globalqss) [inline migration/i7.1z/oracle/202009111500_IDEMPIERE-3981.sql:975]: error: ORA-00907: missing right parenthesis

File: migration/i7.1z/oracle/202009111500_IDEMPIERE-3981.sql [hunk 4]
```diff
@@ -0,0 +1,1036 @@
...
+-- Jun 3,  2019 10:16:32 BRT
+UPDATE AD_Field SET Name='Display Logic', Description='If the Field is displayed, the result determines if the field is actually displayed', Help='format := {expression} [{logic} {expression}]<br>
+expression := @{context}@{operand}{value} or @{context}@{operand}{value}<br>
+logic := {|}|{&}<br>
+context := any global or window context <br>
+value := strings or numbers<br>
+logic operators	:= AND or OR with the previous result from left to right <br>
+operand := eq{=}, gt{&gt;}, le{&lt;}, not{~^!} <br>
+Examples: <br>
+@AD_Table_ID@=14 | @Language@!GERGER <br>
```
Comment (globalqss) [inline migration/i7.1z/oracle/202009111500_IDEMPIERE-3981.sql:114]: lines starting with @ throw error in oracle, like:
  9  SP2-0310: unable to open file "PriceLimitxe>10.sql"
  9  SP2-0310: unable to open file "Namexe>J<br>.sql"

This error happens on the lines starting with:
@AD_Table_ID@=14 | @Language@!GERGER <br>
@PriceLimit@>10 | @PriceList@>@PriceActual@<br>
@Name@>J<br>

They are solved changing those lines to:
'||chr(64)||'AD_Table_ID@=14 | @Language@!GERGER <br>
'||chr(64)||'PriceLimit@>10 | @PriceList@>@PriceActual@<br>
'||chr(64)||'Name@>J<br>
respectively

See for example: migration/i6.2z/oracle/201906261250_IDEMPIERE-3996.sql

There are 9 op these lines in this script

File: org.adempiere.ui.zk/WEB-INF/src/org/adempiere/webui/apps/ProcessParameterPanel.java [hunk 5]
```diff
@@ -209,7 +209,8 @@ public boolean init() {
 					+ "p.FieldLength, p.IsMandatory, p.IsRange, p.ColumnName, "
 					+ "p.DefaultValue, p.DefaultValue2, p.VFormat, p.ValueMin, p.ValueMax, "
 					+ "p.SeqNo, p.AD_Reference_Value_ID, vr.Code AS ValidationCode, "
-					+ "p.ReadOnlyLogic, p.DisplayLogic, p.IsEncrypted, NULL AS FormatPattern, p.MandatoryLogic, p.Placeholder, p.Placeholder2, p.isAutoComplete "
+					+ "p.ReadOnlyLogic, p.DisplayLogic, p.IsEncrypted, NULL AS FormatPattern, p.MandatoryLogic, p.Placeholder, p.Placeholder2, p.isAutoComplete, "
+					+ "'N' AS IsSummarized, p.EntityType, '' AS ValidationCodeLookup "
```
Comment (globalqss) [inline org.adempiere.ui.zk/WEB-INF/src/org/adempiere/webui/apps/ProcessParameterPanel.java:213]: IsSummarized and EntityType are not required here - just ValidationCodeLookup

same thing 12 lines below

**Review by globalqss — CHANGES_REQUESTED** [review]
I did some tests and it seems is working fine.

Please fix the requested problems.

Also in the commit message is good to write:

Co-authored-by: muriloht <muriloht@devcoffee.com.br>

File: org.adempiere.ui.zk/WEB-INF/src/org/adempiere/webui/apps/ProcessParameterPanel.java (hunk 5 above)
Comment (dpansheriya) [reply org.adempiere.ui.zk/WEB-INF/src/org/adempiere/webui/apps/ProcessParameterPanel.java:213]: Thanks @globalqss I missed to clean this one

**Comment by dpansheriya** [comment]
@globalqss all your review comments are taken care in latest commit. 
I did not change previous commit as not sure how github behave as amending commit will create new commit ID. So only added co-author in correction.

**Review by CarlosRuiz-globalqss — APPROVED** [review]

**Pull request merged**